framework/src/media/audio/audio_manager.c

```diff
@@ -159,18 +159,21 @@ static audio_manager_result_t set_audio_volume(audio_card_type_t type, uint8_t v
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
-static audio_manager_result_t find_audio_card_if_empty(audio_card_type_t card_type)
+static audio_manager_result_t init_audio_stream(audio_card_type_t card_type)
```

Comment:
Please think about more about below things.
1. Add one more state, AUDIO_CARD_NONE as a default value
2. g_actual_audio_in_card_id will be founded after card finding finished.
3. If card state is not AUDIO_CARD_NONE , we don't have to set values including mutex init
4. If card finding is finished, we should skip find_audio_card.

Reply:
Updated "if" statement at line 165.
=> This may resolve the first case (two application case).

Added "AUDIO_CARD_NONE" and modified the logic to find an available audio card (=READY state).
=> "find_audio_card" will be called only once and the remaining issues.